linkerd/app/outbound/src/tcp/opaque_transport.rs

```diff
@@ -131,7 +158,17 @@ mod test {
     fn ep(metadata: Metadata) -> Endpoint<()> {
         Endpoint {
             addr: Remote(ServerAddr(([127, 0, 0, 2], 4321).into())),
-            tls: Conditional::None(tls::NoClientTls::NotProvidedByServiceDiscovery),
+            tls: metadata
+                .identity()
+                .map(|id| {
+                    Conditional::Some(tls::ClientTls {
+                        server_id: id.clone(),
+                        alpn: Some(tls::client::AlpnProtocols(vec![PROTOCOL.into()])),
+                    })
+                })
+                .unwrap_or(Conditional::None(
+                    tls::NoClientTls::NotProvidedByServiceDiscovery,
+                )),
```

Comment:
wondering how much boilerplate could be saved by some kind of "`Conditional::from_option`-type helper that just takes a reason for the `None` branch. not a blocker for this PR...